app/models/assignment.rb

```diff
@@ -767,9 +767,7 @@ def average_annotations(ta_id = nil)
   # Returns the groupings of this assignment associated with the given section
   def section_groupings(section)
     groupings.select do |grouping|
-      grouping.inviter.present? &&
-      grouping.inviter.has_section? &&
-      grouping.inviter.section.id == section.id
+      grouping.inviter&.section && grouping.section.id == section.id
```

Comment:
This is better, but can be simplified further. Notice that `nil == x` is `false` whenever `x` is a number.